hazelcast-command-line/src/main/java/com/hazelcast/commandline/HazelcastMember.java

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.commandline;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.ConfigStream;
+import com.hazelcast.config.FileSystemXmlConfig;
+import com.hazelcast.config.FileSystemYamlConfig;
+import com.hazelcast.config.InterfacesConfig;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.config.MemberXmlConfigRootTagRecognizer;
+import com.hazelcast.internal.config.MemberYamlConfigRootTagRecognizer;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+
+import static com.hazelcast.internal.util.StringUtil.isNullOrEmpty;
+
+/**
+ * Class for starting new Hazelcast members
+ */
+final class HazelcastMember {
+    private HazelcastMember() {
+    }
+
+    /**
+     * Creates a server instance of Hazelcast.
+     * <p>
+     * If user sets the system property "print.port", the server writes the port number of the Hazelcast instance to a file.
+     * The file name is the same as the "print.port" property.
+     *
+     * @param args none
+     */
+    public static void main(String[] args)
+            throws Exception {
+        System.setProperty("hazelcast.tracking.server", "true");
+        HazelcastInstance hz = Hazelcast.newHazelcastInstance(config());
+        printMemberPort(hz);
+    }
+
+    private static void printMemberPort(HazelcastInstance hz) throws FileNotFoundException, UnsupportedEncodingException {
+        String printPort = System.getProperty("print.port");
+        if (printPort != null) {
+            try (PrintWriter printWriter = new PrintWriter("ports" + File.separator + printPort, "UTF-8")) {
+                printWriter.println(hz.getCluster().getLocalMember().getAddress().getPort());
+            }
+        }
+    }
+
+    static Config config()
+            throws Exception {
+        String hazelcastConfig = System.getProperty("hazelcast.config");
+        Config config;
+        if (!isNullOrEmpty(hazelcastConfig)) {
+            config = createConfig(hazelcastConfig);
+        } else {
+            config = Config.load();
+        }
+        config.getJetConfig().setEnabled(true);
+        String port = System.getProperty("network.port");
+        if (port != null && !port.equalsIgnoreCase("null")) {
+            config.getNetworkConfig().setPort(Integer.parseInt(port));
+        }
+        String networkInterface = System.getProperty("network.interface");
+        if (networkInterface != null && !networkInterface.equalsIgnoreCase("null")) {
+            config.setProperty("hazelcast.socket.bind.any", "false");
+            InterfacesConfig interfaces = config.getNetworkConfig().getInterfaces();
+            interfaces.setEnabled(true).addInterface(networkInterface);
+        }
+        return config;
+    }
```

Comment:
I have multiple issues with this method, 

- always enables Jet
- `createConfig` uses code to load config which doesn't apply the configuration override (via `-Dhazelcast....`)
- the configs exposed `network.port` and `network.interface` were probably added to support the `--port` and `--interface` parameters and they can be done via configuration override.

We should either just drop this method completely and migrate to config override in the command implementation

Reply:
Yes, we can do that if we agree to break the old behaviour

Reply:
I've removed `config()` method which also fixed overriding configuration with env variables: https://github.com/hazelcast/hazelcast/pull/20262/commits/f3b5f8446e1f5b6b6303e4799e0888affc15b064